mmdet/datasets/dsdl.py

```diff
@@ -0,0 +1,185 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import os
+from typing import List
+
+from dsdl.dataset import DSDLDataset
```

Comment:
To avoid compatibility issues, please import like this
```python
try:
    from dsdl.dataset import DSDLDataset
except ImportError:
    DSDLDataset = None
```

And then check the installation when using it:

```python
        if DSDLDataset is None:
            raise RuntimeError(
                'Package dsdl is not installed. Please run "pip install dsdl".')
```